### PR TITLE
Handle substeps in LTS error control

### PR DESCRIPTION
**Pull request by wthrowe**
Preparation for adding LTS substep methods.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Time/StepperErrorEstimate.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+
+#include "Time/Time.hpp"
+
+template <typename T>
+struct StepperErrorEstimate {
```
Comment (nilsdeppe) [inline src/Time/StepperErrorEstimate.hpp:14]: Could you please add doxygen documentation to this?

File: src/Time/StepperErrorEstimate.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  Time step_time{};
+  TimeDelta step_size{{}, 0};
+  size_t order{};
+  T error{};
+
+  void pup(PUP::er& p) {
```
Comment (nilsdeppe) [inline src/Time/StepperErrorEstimate.hpp]: Move out of line?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, a couple minor suggestions you can squash immediately

**Comment by wthrowe** [comment]
Done.  Also changed the initializer for `TimeDelta step_size{{}, 0};` to plain `{}` because that weirdness was left over from an earlier attempt.

File: src/Time/StepChoosers/ErrorControl.hpp [hunk 3]
```diff
@@ -182,56 +174,56 @@ class ErrorControl : public StepChooser<StepChooserUse>,
...
                   pow(1.0 / std::max(l_inf_error, 1e-14), alpha_factor) *
                   pow(std::max(previous_l_inf_error, 1e-14), beta_factor),
               min_factor_, max_factor_);
     }
-    return std::make_pair(new_step, l_inf_error <= 1.0);
+    // If the error is out-of-date we can still reject a step, but it
+    // won't improve the reported error, so make sure to only do it
+    // if we actually request a smaller step.
+    return std::make_pair(new_step,
+                          new_step >= previous_step or l_inf_error <= 1.0);
```
Comment (wthrowe) [inline src/Time/StepChoosers/ErrorControl.hpp]: Wait, this is backwards.

Comment (wthrowe) [reply src/Time/StepChoosers/ErrorControl.hpp]: Actually, no, it's the right way, but it does need an `abs`.

Comment (wthrowe) [reply src/Time/StepChoosers/ErrorControl.hpp]: Pushed a fixup for this and updated the test so it would have been caught.

**Comment by wthrowe** [comment]
And added a small commit fixing a preexisting bug in the test.

**Review by nilsdeppe — COMMENTED** [review]
LGTM! Please go ahead and squash!

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**